consumer/services/ucam.py

```diff
@@ -0,0 +1,121 @@
+import os
+from datetime import datetime
+from typing import List, Optional
+
+import requests
+
+from consumer.schemas.ucam import (
+    Device,
+    DevicePatient,
+    DiseaseType,
+    Patient,
+    PatientDevice,
+)
+
+
+def __ucam_access_token() -> str:
+    """Obtain (or refresh) an access token."""
+    now = int(datetime.utcnow().timestamp())
+    last_created = int(os.getenv("UCAM_ACCESS_TOKEN_GEN_TIME", 0))
+    # Refresh the token every 1 day, i.e., below 7 day limit.
+    token_expired = (last_created + (60 * 60 * 24)) <= now
+
+    if token_expired:
+        request = {
+            "Username": os.getenv("UCAM_USERNAME"),
+            "Password": os.getenv("UCAM_PASSWORD"),
+        }
+
+        response = requests.post(f"{os.getenv('UCAM_URI')}/user/login", json=request)
+        response.raise_for_status()
+        result: dict = response.json()
+        access_token = result["token"]
+
+        os.environ["UCAM_ACCESS_TOKEN"] = access_token
+        os.environ["UCAM_ACCESS_TOKEN_GEN_TIME"] = str(now)
+
+    return os.getenv("UCAM_ACCESS_TOKEN")
+
+
+def response(request_url: str) -> Optional[dict]:
+    """
+    Performs GET request on the UCAM API
+    NOTE: requests automatically converts null to None
+    """
+    headers = {"Authorization": f"Bearer {__ucam_access_token()}"}
+    url = f"{os.getenv('UCAM_URI')}{request_url}"
+
+    response = requests.get(url, headers=headers)
+    response.raise_for_status()
+
+    # possibly no result
+    if response.status_code == 204:
+        return None
```

Comment:
I have requested, but no response yet. The difference is most likely since patients are unique and (unfortunately due to the past use of UCAM) devices and VTT are not. But either way - have requested it but we can work with this at the moment.